### ISSUE TITLE
External MongoDB Not Being Used When Defining environment variables

### ISSUE THREAD

**Issue by rayray221** [jacobalberty/unifi-docker#401]
### Host operating system

Fedora 31

### What tag are you using

latest

### What complete docker command or docker-compose.yml do you use to launch the container (omitting sensitive values)?
```
version: '2.3'
services:
  controller:
    image: "jacobalberty/unifi:latest"
    container_name: test_controller
    init: true
    restart: always
    volumes:
      - /unifi:/unifi
    user: unifi
    sysctls:
      net.ipv4.ip_unprivileged_port_start: 0
    environment:
      DB_URI: mongodb://USERNAME:PASSWOR@10.10.10.10:27017/unifi?authSource=admin
      STATDB_URI: mongodb://USERNAME:PASSWORD@10.10.10.10:27017/unifi_stat?authSource=admin
      DB_NAME: unifi
    ports:
      - "3478:3478/udp" # STUN
      - "6789:6789/tcp" # Speed test
      - "8080:8080/tcp" # Device/ controller comm.
      - "8443:8443/tcp" # Controller GUI/API as seen in a web browser
      - "8880:8880/tcp" # HTTP portal redirection
      - "8843:8843/tcp" # HTTPS portal redirection
      - "10001:10001/udp" # AP discovery
  logs:
    image: bash
    container_name: logs
    depends_on:
      - controller
    command: bash -c 'tail -F /unifi/log/*.log'
    restart: always
    volumes:
      - /unifi/log:/unifi/log
```

### What do you expect to happen?
Unifi network controller uses external MongoDB defined in the environment variables.

### What actually happens?
Local Mongo DB is created and used

Note:  I can connect to MongoDB using the defined URIs in the environment variables.  Also, I have created and destroyed this a bunch of times and once or twice it did use the external DB.....I'm guessing it is a timing issue or sorts.  I can also add the four config options to the system.properties (and nothing else) and start up the containers for the first time...this also works and the external DB is used.

Logs just show that the local DB is being used:
```
[2020-12-22T17:03:02,010] <localhost-startStop-1> INFO  db     - waiting for db connection...
[2020-12-22T17:03:03,514] <localhost-startStop-1> INFO  db     - Connecting to mongodb://localhost:27117
[2020-12-22T17:03:05,589] <localhost-startStop-1> INFO  db     - setting mongodb features compatibility version to 3.6
[2020-12-22T17:03:45,835] <localhost-startStop-1> INFO  db     - Connecting to mongodb://localhost:27117
```

Happy to provide additional information, not sure what else to try.
Thanks 
Ray

**Comment by rayray221** [comment]
Interesting enough....the first time I start the containers, it uses the local DB in the container image....when I shutdown and restart the containers, it then uses the External MongoDB....

**Comment by Jackman3005** [comment]
@rayray221 I too see that the first time it starts (if unifi directory is deleted for example) then it uses localhost for db connection and then subsequent calls use Mongodb. I have been investigating this in regard to the fact that my controller is not accessible after restarting via `docker-compose down && docker-compose up`

See [semi-related issue](https://github.com/jacobalberty/unifi-docker/issues/165) on not being able to login after restarting...

**Comment by camh-** [comment]
I have just be diagnosing the same issue. I believe it is because the `docker-entrypoint.sh` script creates the `${DATADIR}` directory after it writes the `systems.properties` file. The first time the container image runs on an empty volume for the unifi volume, the data dir does not exist so it fails to write the properties file, so does not tell the controller not to start mongo db. The second time it runs, it is able to write the properties file, so the controller does not start mongo db.

**Comment by camh-** [comment]
This should have been fixed in #419 and can be closed
